Objects/typeobject.c

```diff
@@ -11285,6 +11294,83 @@ update_all_slots(PyTypeObject* type)
     }
 }
 
+int
+_PyType_InitSlotDefsNameCounts(PyInterpreterState *interp)
+{
+    if (interp != interp->runtime->interpreters.main) {
+        return 0;
+    }
+    PyObject *bytes = NULL;
+    PyObject *cache = PyDict_New();
+    if (!cache) {
+        return -1;
+    }
+
+    pytype_slotdef *p;
+    Py_ssize_t idx = 0;
+    for (p = slotdefs; p->name_strobj; p++, idx++) {
+        Py_hash_t hash = _PyObject_HashFast(p->name_strobj);
+        if (hash == -1) {
+            goto error;
+        }
+        assert (idx < 255);
+
+        if (_PyDict_GetItemRef_KnownHash_LockHeld((PyDictObject *)cache,
```

Comment:
_PyType_InitSlotDefsNameCounts() is called only once, so it's no needed to micro-optimize it. Just use PyDict_GetItemRef() and PyDict_SetItem().